lrpar/src/lib/parser.rs

```diff
@@ -396,12 +437,15 @@ where
                             spans_uw.truncate(pop_idx - 1);
                             spans_uw.push(span);
 
-                            let v = AStackType::ActionType(self.actions[usize::from(pidx)](
+                            let (x, param_tmp) = self.actions[usize::from(pidx)](
                                 ridx,
                                 self.lexer,
                                 span,
                                 astack_uw.drain(pop_idx - 1..),
-                            ));
+                                param,
+                            );
+                            param = param_tmp;
```

Comment:
So here is one of the places we regain ownership via returning Param from action.